app/services/network_cache/fetcher.rb

```diff
@@ -8,10 +8,12 @@ def self.from(testing_ground, opts = {})
 
     #
     # Fetches all cache and sets is as the load attribute for a node
-    def fetch
+    def fetch(nodes = [])
       tree_scope.each do |network|
         network.nodes.each do |node|
```

Comment:
We could avoid iterating through all the nodes in the network like so:

```ruby
nodes.map { |key| network.node(key) }.compact.each do |node|
  network.node(key).set(:load, read(network.carrier, node.key) || [])
end
```

Not a big deal, but it's slightly more efficient in large networks. The `map.compact` step is required in case the request specifies a node which does not exist. *Ideally* we'd reject the request (with `400 Bad Request` or `422 Unprocessable Entity`) before getting to this point, in which case the code becomes simpler:

```ruby
nodes.each do |key|
  network.node(key).set(:load, read(network.carrier, node.key) || [])
end
```